src/middlewared/middlewared/plugins/virt/instance_device.py

```diff
@@ -198,11 +199,20 @@ async def generate_device_name(self, device_names: list[str], device_type: str)
             i += 1
         return name
 
-    async def __validate_device(self, device, schema, verrors: ValidationErrors):
+    async def __validate_device(self, device, schema, verrors: ValidationErrors, instance: str = None):
         match device['dev_type']:
             case 'PROXY':
-                verror = await self.middleware.call('port.validate_port', schema, device['source_port'])
-                verrors.extend(verror)
+                # We want to make sure there are no other instances using that port
+                ports = await self.middleware.call('port.ports_mapping')
```

Comment:
Do we need to always validate `source_port` even if it doesn't change?

Reply:
Good point, will fix